src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

```diff
@@ -77,11 +77,24 @@ public void visitToken(DetailAST ast) {
             final DetailAST lastGroupAST =
                 switchAST.getLastChild().getPreviousSibling();
 
-            if (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
-                || defaultGroupAST.getColumnNo()
-                    != lastGroupAST.getColumnNo()) {
+            if (!containCaseBeforeOrAfterDefault(ast)
+                    && (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
+                        || defaultGroupAST.getColumnNo()
+                            != lastGroupAST.getColumnNo())) {
                 log(ast, MSG_KEY);
             }
         }
     }
+
+    /**
+     * Examines ast to see if it is surrounded by a 'case' on either side.
+     *
+     * @param ast - root node.
+     * @return true if default is surrounded by a 'case' on either side or false.
+     */
+    private boolean containCaseBeforeOrAfterDefault(DetailAST ast) {
+        return ast.getPreviousSibling() != null
+                || ast.getNextSibling() != null
+                && ast.getNextSibling().getType() == TokenTypes.LITERAL_CASE;
```

Comment:
Reviewed the code in my project again which made me think. Since all cases have to be defined and a missing case is an error, the default case should log an error. However, there is still a `DEFAULT == -1` case. I guess that's the reason why default has been used not at the end. Because there is a `DEFAULT` case but it shouldn't be the `default` case. It's a little complicated to explain without going into detail but in source code, it would look like this:

```java
// The code we should use in our project instead
switch (id):
    case x:
    case DEFAULT: // DEFAULT case we should've used
         show(X);
         break;
    case y:
        show(Y);
        break;
    default: // real default case
        log(error);
        break;
```

So we have to improve our code instead of using a new option. :flushed:

This would've been the real code https://gist.github.com/robertpainsi/259a742c6059394ed6d13cc7cabaf71e